packages/dds/task-manager/src/taskManager.ts

```diff
@@ -312,25 +317,55 @@ export class TaskManager extends SharedObject<ITaskManagerEvents> implements ITa
         return lockAcquireP;
     }
 
-    public abandon(taskId: string) {
-        if (!this.connected) {
-            throw new Error(`Attempted to abandon in disconnected state: ${taskId}`);
+    public subscribeToTask(taskId: string) {
+        if (this.subscribed(taskId)) {
+            return;
         }
 
-        // Nothing to do if we're not at least trying to get the lock.
-        if (!this.queued(taskId)) {
-            return;
+        const submitVolunteerOp = () => {
+            this.submitVolunteerOp(taskId);
+        };
+
+        const disconnectHandler = () => {
+            // Wait to be connected again and then re-submit volunteer op
+            this.connectionWatcher.once("connect", submitVolunteerOp);
+        };
+
+        const checkIfAbandoned = (eventTaskId: string) => {
+            if (eventTaskId !== taskId) {
+                return;
+            }
+
+            this.abandonWatcher.off("abandon", checkIfAbandoned);
+            this.connectionWatcher.off("disconnect", disconnectHandler);
+            this.connectionWatcher.off("connect", submitVolunteerOp);
+
+            this.subscribedTasks.set(taskId, false);
```

Comment:
Any thoughts on using a Set instead of a Map, and just deleting the task from the Set when abandoning?

Reply:
Good call, forgot about sets 😅